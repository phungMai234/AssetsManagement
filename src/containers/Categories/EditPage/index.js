import React, { useCallback } from 'react';
import ModalAddCategory from '../ModalAddCategory';
import * as Yup from 'yup';
import { Formik } from 'formik';
import db from 'database';
import useAlert from 'hooks/useAlert';

const EditPage = ({ onCancel, record }) => {
  const { setAlert } = useAlert();

  const initialValues = {
    name: record?.name || '',
    manager: record?.manager || '',
    note: record?.note || '',
  };

  const validationSchema = Yup.object({
    name: Yup.string().required('Trường bắt buộc').max(255, 'Không quá 255 kí tự'),
    manager: Yup.string().trim().required('Trường bắt buộc').max(255, 'Không quá 255 kí tự'),
    note: Yup.string().trim().max(2000, 'Không quá 2000 kí tự'),
  });

  const handleSubmit = useCallback(
    (values) => {
      db.collection('categories')
        .doc(record?.id)
        .update({ ...values })
        .then(() => {
          onCancel();
          setAlert({ status: 'success', message: 'Cập nhật chỉnh sửa thành công' });
        })
        .catch(() => {
          setAlert({ status: 'danger', message: 'Đã xảy ra lỗi hệ thống. Vui lòng thử lại!' });
        });
    },
    [onCancel, record, setAlert],
  );

  return (
    <Formik initialValues={initialValues} validationSchema={validationSchema} onSubmit={handleSubmit}>
      {(props) => <ModalAddCategory isEdit onHide={onCancel} {...props} />}
    </Formik>
  );
};

export default EditPage;
