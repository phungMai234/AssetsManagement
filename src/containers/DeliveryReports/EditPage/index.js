import React, { useContext, useMemo } from 'react';

import FormContainer from '../FormContainer';
import { useParams } from 'react-router-dom';
import Loading from 'components/Loading';
import { DeliveryReportContext } from 'contexts/DeliveryReportContext';

const EditPage = () => {
  const { id } = useParams();

  const { data, loading } = useContext(DeliveryReportContext);

  const restructureData = useMemo(() => {
    if (loading) {
      return {};
    }

    let order = data.find((e) => e.id === id);

    return order;
  }, [data, id, loading]);

  if (loading || !data) {
    return <Loading />;
  }

  return (
    <FormContainer
      data={{
        ...restructureData,
        user_name: { name: restructureData?.user_name, label: restructureData?.user_name },
        orderDetails: restructureData?.order_details,
      }}
    />
  );
};

export default EditPage;
