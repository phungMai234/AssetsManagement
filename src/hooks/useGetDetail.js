import { useState, useEffect, useCallback } from 'react';

import db from '../database';

const useGetDetail = ({ nameCollection, id, ignoreCancel = false }) => {
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(false);
  const [forceRequest, setForceRequest] = useState(false);

  useEffect(() => {
    setLoading(true);

    db.collection(nameCollection)
      .doc(id)
      .onSnapshot((querySnapshot) => {
        setData(querySnapshot.data());
        setLoading(false);
      });
  }, [forceRequest, ignoreCancel, nameCollection, id]);

  const force = useCallback(() => {
    setForceRequest({});
  }, []);

  return { loading, data, force };
};

export default useGetDetail;
