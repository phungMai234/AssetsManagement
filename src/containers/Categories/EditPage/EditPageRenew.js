import React from 'react';

import FormContainer from '../FormContainer';
import { useParams } from 'react-router-dom';
import useGetDetail from 'hooks/useGetDetail';
import { useGetDetailProductLine } from 'hooks/useGetDetailProductLine';
import Loading from 'components/Loading';

const EditPage = () => {
  const { id } = useParams();

  const { data, loading } = useGetDetail({ nameCollection: 'categories', id: id });
  const { dataProduct, loading: loadingProductLine } = useGetDetailProductLine({ id: id });

  if (loading || loadingProductLine) {
    return <Loading />;
  }
  return <FormContainer data={{ ...data, orderDetails: dataProduct, id: id }} />;
};

export default EditPage;
