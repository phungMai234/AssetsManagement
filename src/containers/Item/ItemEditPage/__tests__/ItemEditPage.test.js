import React from 'react';

import { render, findByText } from '@testing-library/react';

import EditPage from '../ItemListPage';
import useGetDetail from 'hooks/useGetDetail';
import reactRouter from 'react-router-dom';

jest.mock('hooks/useGetDetail');
jest.mock('react-router-dom', () => ({
  useParams: jest.fn().mockReturnValue({ id: '123' }),
  useHistory: jest.fn(),
}));

describe('Item Edit Page', () => {
  test('should render correctly', () => {
    useGetDetail.mockReturnValue({
      loading: false,
      data: {
        data: [
          {
            classroom: { id: 418, name: 'lop0', student_age_range: '0歳児' },
            id: 1,
            name: 'nghia0',
            name_hira: 'てすと',
          },
        ],
      },
    });
    const container = render(<EditPage />);
    expect(container.container).toMatchSnapshot();
  });

  test('should render correctly with loading', () => {
    useGetDetail.mockReturnValue({
      loading: true,
      data: {},
    });
    const wrapper = render(<EditPage />);
    const textLoading = wrapper.findByText('Loading...');
    expect(textLoading).toBeTruthy();
    expect(wrapper.container).toMatchSnapshot();
  });
});
