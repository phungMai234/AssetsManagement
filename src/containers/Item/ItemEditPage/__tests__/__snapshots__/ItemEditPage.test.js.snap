// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Edit Page should render correctly 1`] = `
<div>
  <div
    class="sc-bdfBwQ jvFxvm"
  >
    <div
      class="sc-eCssSg fJaZbo"
    >
      <div
        class="spinner-border"
        role="status"
      >
        <span
          class="sr-only"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Item Edit Page should render correctly with loading 1`] = `
<div>
  <div
    class="sc-eCssSg fJaZbo"
  >
    <div
      class="spinner-border"
      role="status"
    >
      <span
        class="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
